### DISCUSSION TITLE
How to use it with nextjs

### DISCUSSION THREAD

**Discussion by ObaidAshiq** [nfriedly/express-rate-limit#439] [General]
I want to rate limit my api routes in next js
How can i do so?

**Comment by nfriedly** [comment]
express-rate-limit is not out-of-the-box compatible with next.js.

You could try using something like [next-connect](https://www.npmjs.com/package/next-connect) to make the two work together, or add a custom Express web server that runs express-rate-limit and then forwards requests on to next. Here are a couple of articles that talk about that: 
* [Next.js: invite Express.js as Middleware](https://medium.com/@obulareddyveera/next-js-invite-express-js-as-middleware-ea5e7bb494f0)
* [Integrating Express.js Server with Next.js-13: Unleash the Power of Combined Frameworks ](https://dev.to/wilmela/integrating-expressjs-server-with-nextjs-13-unleash-the-power-of-combined-frameworks-10kc))

I haven't tried any of these, so I can't say for sure how well it will work.

Alternatively, a quick search turned up a couple of rate limiters that were made for next.js:
* https://www.npmjs.com/package/next-rate-limit
* https://www.npmjs.com/package/@lidofinance/next-ip-rate-limit

I'd probably recommend trying those out before attempting to wrangle express-rate-limit into working with next.js.

**Reply by ObaidAshiq** [reply]
will try them and
